sshoot/profile.py

```diff
@@ -42,7 +41,7 @@ def from_config(cls, config: Dict[str, Any]):
 
     def update(self, config: Dict[str, Any]):
         """Update the profile from the specified config."""
-        field_names = self._field_names()
+        field_names = self._fields().keys()
```

Comment:
this should be `field_names = list(self._fields())`, since it's traversed multiple times below

Reply:
I updated this definition